instance/instance.go

```diff
@@ -107,32 +102,51 @@ func Create(domain string, locale string, apps []string) (*Instance, error) {
 		StorageURL: domainURL.String(),
 	}
 
-	err := i.Create()
+	var err error
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I find it cleaner for the function to always return result XOR error, but never both.